rackcity/views/it_instance_views.py

```diff
@@ -1,19 +1,48 @@
 # from rest_framework.parsers import JSONParser
 from django.http import HttpResponse, JsonResponse
 from rackcity.models import ITInstance
-from rackcity.api.serializers import ITInstanceSerializer
+from rackcity.api.serializers import (
+    ITInstanceSerializer,
+    RecursiveITInstanceSerializer,
+)
+from rest_framework.decorators import permission_classes, api_view
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
+from rest_framework.parsers import JSONParser
+from rest_framework.pagination import PageNumberPagination
+from http import HTTPStatus
 
 
+@api_view(['GET'])
+@permission_classes([IsAuthenticated])
 def instance_list(request):
     """
     List all instances.
     """
     if request.method == 'GET':
         instances = ITInstance.objects.all()
-        serializer = ITInstanceSerializer(instances, many=True)
+        serializer = RecursiveITInstanceSerializer(instances, many=True)
         return JsonResponse(serializer.data, safe=False)
 
 
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def instance_page(request):
+    """
+    List a page of instances.
+    """
+    instances = ITInstance.objects.all()
+    paginator = PageNumberPagination()
+    paginator.page_size = 10
+    page_of_instances = paginator.paginate_queryset(instances, request)
```

Comment:
I wish it was more clear that this gets the page number that was specified in the request but I'm not sure how to do that. Do you think it's worth an inline comment?

Reply:
I agree. Will add comment for now